nemoguardrails/library/autoguard/flows.co

```diff
@@ -0,0 +1,43 @@
+define flow autoguard check input
+  $input_result = execute autoguard_input_api(show_autoguard_message=True)
+
+  if $input_result["guardrails_triggered"]
+    $autoguard_input_response = $input_result['combined_response']
+    bot refuse to respond
+    stop
+
+define flow autoguard check output
+  $output_result = execute autoguard_output_api(show_autoguard_message=True)
+
+  if $output_result["guardrails_triggered"]
+    bot refuse to respond
+    stop
+  else
+    $pii_message_output = $output_result["pii_fast"]["response"]
+    if $output_result["pii_fast"]["guarded"]
+      bot respond pii output
+      stop
+
+define flow autoguard factcheck input
+  execute autoguard_retrieve_relevant_chunks_input
+  $input_result = execute autoguard_factcheck_input_api
+
+define flow autoguard factcheck output
+  $output_result = execute autoguard_factcheck_output_api
+  if $input_result < 0.5
+    bot inform autoguard factcheck input violation
```

Comment:
hmm @rahm-decypher @abhijitpal1247 

Not sure if i'm reading this correctly, my interpretation is that it's checking input prompt for factchecking purposes. Does that make sense?

Reply:
fact checking should be on the model output and the list of evidence documents. please change the flow if that is not the case

Reply:
we have two separate interfaces for user input and bot response respectively. Should I make the factheck flow a blocking one as we have the normal autoguard flow in blocking mode. @KimiJL @rahm-decypher ?

Reply:
The idea should be that factcheck isn't run on the user input, and only on the bot response, regardless of blocking or non-blocking